test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1392,12 +1353,133 @@ def test_incr_decr(self):
-                                         starttime=int(self.t))
-        except PtlLogMatchError, e:
-            self.logger.info(
-                "Number of job equivalence classes message\
-                not present when there are no jobs as expected")
-            self.assertNotEqual(e.rc, 0)
+        self.scheduler.log_match("Number of job equivalence classes",
+                                 max_attempts=10,
+                                 starttime=self.t,
+                                 existence=False)
+        self.logger.info(
+            "Number of job equivalence classes message" +
+            "not present when there are no jobs as expected")
+
+    def test_server_queue_limit(self):
+        """
+        Test with mix of hard and soft limits
+        on resources for users and groups
+        """
+
+        # Update sched_config
+        a = {'preempt_prio':
+             'express_queue, normal_jobs, server_softlimits,\
+             queue_softlimits,server_softlimits+queue_softlimits'}
+        self.scheduler.set_sched_config(a)
+
+        # Create workq2
+        self.server.manager(MGR_CMD_CREATE, QUEUE,
+                            {'queue_type': 'e', 'started': 'True',
+                             'enabled': 'True'}, id='workq2')
+
+        # Set ncpus to 50
+        self.server.manager(
+            MGR_CMD_SET, NODE,
+            {'resources_available.ncpus': '50'},
+            id=self.mom.shortname)
+
+        # Set queue limit
+        a = {
+            'max_run': '[o:PBS_ALL=100],[g:PBS_GENERIC=20],\
+                       [u:PBS_GENERIC=20],[g:tstgrp01 = 8],[u:pbsuser1=10]'}
+        self.server.manager(MGR_CMD_SET, QUEUE,
+                            a, id='workq2')
+
+        a = {'max_run_res.ncpus':
+             '[o:PBS_ALL=100],[g:PBS_GENERIC=50],\
+             [u:PBS_GENERIC=20],[g:tstgrp01=13],[u:pbsuser1=12]'}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id='workq2')
+
+        a = {'max_run_res_soft.ncpus':
+             '[o:PBS_ALL=100],[g:PBS_GENERIC=30],\
+             [u:PBS_GENERIC=10],[g:tstgrp01=10],[u:pbsuser1=10]'}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, id='workq2')
+
+        # Create server limits
+        a = {
+            'max_run': '[o:PBS_ALL=100],[g:PBS_GENERIC=50],\
+            [u:PBS_GENERIC=20],[g:tstgrp01=13],[u:pbsuser1=13]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        a = {'max_run_soft':
+             '[o:PBS_ALL=50],[g:PBS_GENERIC=25],[u:PBS_GENERIC=10],\
+             [g:tstgrp01=10],[u:pbsuser1=10]'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Turn scheduling off
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'false'})
+
+        # Submit jobs as pbsuser1 from group tstgrp01 in workq2
+        a = {'Resource_List.select': '1:ncpus=1',
+             'group_list': TSTGRP1, ATTR_q: 'workq2'}
+        J = Job(TEST_USER1, attrs=a)
+        for n in range(10):
+            self.server.submit(J)
```

Comment:
Use the submit_jobs() method instead of directly submitting jobs (and for the rest of the test)